### PR TITLE
mConfigurables introduced to the checks, new check on laser

### PR DESCRIPTION
**Pull request by lauraser**
1. PadCalibration changed to include configurable parameters for check (additional line deletes are approved by Thomas)
2. Checks updated, so that user can choose which c_ROCs canvas has to be checked for empty pads, multiple c_ROCs can be passed as well
3. Limits of medium and bad quality for padcalibration check are now configurables
4. Additional json file with checks on laser calibration trending

### CODE REVIEW

**Comment by Barthelemy** [comment]
@wiechula do you want to review or do we go ahead ?

**Comment by wiechula** [comment]
@Barthelemy , I'll have a look and approve if it's fine.

File: Modules/TPC/run/tpcQCPadCalibration.json [hunk 1]
```diff
@@ -115,8 +106,17 @@
...
+        }],
+        "stopTrigger_comment": [ "MOsNames are required for code to work, runs check of empty pads, should be the same as MOs, the MOs must be separated by comma only, no spaces" ],
+        "checkParameters": {
+          "mediumQualityPercentageOfWorkingPads" : "0.7",
+          "badQualityPercentageOfWorkingPads" : "0.4",
+          "mediumQualityNoiseMean" : "1.25",
+          "badQualityNoiseMean" : "1.5",
+          "MOsNames2D" : "c_ROCs_Pedestals_2D,c_ROCs_Noise_2D"
+        }
+
```
Comment (wiechula) [inline Modules/TPC/run/tpcQCPadCalibration.json:119]: How about the PID and Track cluster check. Wasn't the idea to generalize the PadCalibrationCheck to be able to deal with all? If so, should the checks be added in this json?

File: Modules/TPC/include/TPC/PadCalibrationCheck.h [hunk 2]
```diff
@@ -46,6 +46,11 @@ class PadCalibrationCheck : public o2::quality_control::checker::CheckInterface
   std::vector<float> mNoiseMean;
   std::vector<float> mNoiseStdDev;
   std::vector<int> mNoiseNonZeroEntries;
+  std::vector<std::string> mMOsToCheck2D;
+  double mMediumQualityLimit;
+  double mBadQualityLimit;
+  double mMediumQualityLimitNoiseMean;
+  double mBadQualityLimitNoiseMean;
```
Comment (wiechula) [inline Modules/TPC/include/TPC/PadCalibrationCheck.h:53]: I think the idea was to generalize the class to be able to use it for all PadCalibration objects. So I would assume that no specific reference to e.g. `Noise` should appear.

File: Modules/TPC/src/PadCalibrationCheck.cxx [hunk 3]
```diff
@@ -30,8 +30,29 @@
...
+  }
+  if (auto param = mCustomParameters.find("badQualityPercentageOfWorkingPads"); param != mCustomParameters.end()) {
+    mBadQualityLimit = ::atof(param->second.c_str());
+  }
+  if (auto param = mCustomParameters.find("mediumQualityNoiseMean"); param != mCustomParameters.end()) {
+    mMediumQualityLimitNoiseMean = ::atof(param->second.c_str());
+  }
+  if (auto param = mCustomParameters.find("badQualityNoiseMean"); param != mCustomParameters.end()) {
+    mBadQualityLimitNoiseMean = ::atof(param->second.c_str());
+  }
```
Comment (wiechula) [inline Modules/TPC/src/PadCalibrationCheck.cxx:46]: use `std::atof`. Remove dedicated reference to `Nosie`?

File: Modules/TPC/src/PadCalibrationCheck.cxx [hunk 4]
```diff
@@ -185,23 +206,23 @@ std::string PadCalibrationCheck::getAcceptedType() { return "TCanvas"; }
 void PadCalibrationCheck::beautify(std::shared_ptr<MonitorObject> mo, Quality)
 {
   auto moName = mo->getName();
-  if (moName == "c_Sides_Noise" || moName == "c_ROCs_Noise_1D" || moName == "c_ROCs_Pedestal_2D") {
+  if (auto it = std::find(mMOsToCheck2D.begin(), mMOsToCheck2D.end(), moName); it != mMOsToCheck2D.end() || moName == "c_Sides_Noise" || moName == "c_ROCs_Noise_1D") {
```
Comment (wiechula) [inline Modules/TPC/src/PadCalibrationCheck.cxx:209]: Why still dedicated check for `Noise`?

**Review by wiechula — CHANGES_REQUESTED** [review]

**Comment by stheckel** [comment]
Hi @lauraser 
please check Jens' comments and update this PR.
Also note, there have been some small changes in some of the files, see [this commit](https://github.com/AliceO2Group/QualityControl/commit/ac1917968e4abdd76129415dba710e9ad6753e0a), leading to merge conflicts here. So please rebase and solve these conflicts before pushing again.
Thanks and cheers,
Stefan

File: Modules/TPC/run/tpcQCPadCalibration.json (hunk 1 above)
Comment (lauraser) [reply Modules/TPC/run/tpcQCPadCalibration.json]: This check is supossed to work on all tasks which provide the canvas with 72 pads. It is already tested in PadCalibration and RawDigits (please, see corresponding json files in this PR). So I call the same check in all tasks which requires such generalized check.

**Comment by lauraser** [comment]
Dear @wiechula,
I separated the code into two tasks. Now the generalised check for empty pads is in CheckForEmptyPads.cxx class and the noise check is separately in the specific check for pad calibration task PadCalibrationCheck.cxx. The CheckForEmptyPads can be called from any json file and is generalised for all tcanvases with 72 pads.
Cheers
Laura

**Review by wiechula — APPROVED** [review]
@lauraser , I propose to get this in for the moment and re-discuss later.

**Pull request merged**